src/node_worker.cc

```diff
@@ -132,7 +132,10 @@ class WorkerThreadData {
  public:
   explicit WorkerThreadData(Worker* w)
     : w_(w) {
-    CHECK_EQ(uv_loop_init(&loop_), 0);
+    if (uv_loop_init(&loop_) != 0) {
+      w->custom_error_ = "ERR_WORKER_OUT_OF_MEMORY";
```

Comment:
Can we use `uv_err_name_r ` to get error codes rather than hard coding a `"ERR_WORKER_OUT_OF_MEMORY"`? Besides, it doesn't seem to be guaranteeing that the error has to be `"ERR_WORKER_OUT_OF_MEMORY"`.

Reply:
@legendecas - the issue with that approach is that we cannot map the return value of `uv_err_name_r` to a pre-existing error macro from https://github.com/nodejs/node/blob/67e067eb0658281b647ff68a5a9e64ea2cfdb706/lib/internal/errors.js . The only use of `uv_err_name_r` in node.js as I can see is from inspector, which is directly printing the error on to the console. Here , I guess we need to pass the error to the JS error call back, and the callee will expect a valid error type, as prescribed in the documented error type. Any guidance?

Reply:
could do `ERR_WORKER_INIT_FAILED: %s`